osu.Game/Skinning/PoolableSkinnableSample.cs

```diff
@@ -129,6 +129,8 @@ private void updateSample()
         /// </summary>
         public void Play()
         {
+            FlushPendingSkinChanges();
```

Comment:
Not sure if I'm understanding correctly, but this seems to be an opportunity to get rid of this thing:

https://github.com/ppy/osu/blob/88700c85fd22a74750fcdac223b661c7774e22f9/osu.Game/Skinning/PoolableSkinnableSample.cs#L80-L85

which is trying to compensate for the same issue this PR is tackling by manually subscribing to `SourceChanged` - the innards of that methods should *theoretically* be safe to move to `SkinChanged()` itself now.

Reply:
Yeah, we could most likely get rid of this. This would of course then make the [`FlushPendingSkinChanges`](https://github.com/ppy/osu/pull/22872/files#r1142639585) necessary.

Reply:
I think I would prefer doing that as part of this pull. Having two workarounds for the same issue live next to each other isn't sitting very well with me.